src/components/MyAccount/Settings/SettingsInputForm.tsx

```diff
@@ -42,9 +43,26 @@ const SettingsInputForm = ({
 
   const [tempInputs, setTempInputs] = useState([...inputs])
 
+  const inputRefs = useRef<Array<TextInputRefType | null>>([])
+  const editingRef = useRef<HTMLButtonElement | null>()
+
+  const focusLastInput = () => {
+    const lastIndex = tempInputs.length - 1
+    if (inputRefs.current[lastIndex]) {
+      inputRefs.current[lastIndex]?.focus()
```

Comment:
I was going to ask if it was possible to use the `Map` function like the code example in the React docs resource. It'll work but I imagine this part here would make it difficult to keep track of the previous element to focus?

Reply:
With an Array we get the order maintained for free, which is crucial here. FWIW, the docs do say "This lets you maintain your own _array_ or a [Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map)".